libsplinter/src/biome/rest_api/actix/token.rs

```diff
@@ -0,0 +1,179 @@
+                            .into_future();
+                    }
+                };
+
+                if refresh_token != refresh_token_from_db {
+                    return HttpResponse::Forbidden()
+                        .json(ErrorResponse::forbidden("Invalid Refresh Token"))
+                        .into_future();
+                }
+
+                match validate_claims(
+                    &refresh_token,
+                    &refresh_token_secret_manager,
+                    &refresh_token_validation,
+                ) {
+                    AuthorizationResult::Authorized(_) => (),
+                    AuthorizationResult::Unauthorized(msg) => {
+                        if let Err(err) = refresh_token_store.remove_token(&claims.user_id()) {
+                            error!("Failed to delete refresh token {}", err);
+                            return HttpResponse::InternalServerError()
+                                .json(ErrorResponse::internal_error())
+                                .into_future();
+                        } else {
+                            return HttpResponse::Unauthorized()
+                                .json(ErrorResponse::unauthorized(&msg))
+                                .into_future();
+                        }
+                    }
+                    AuthorizationResult::Failed => {
+                        return HttpResponse::InternalServerError()
+                            .json(ErrorResponse::internal_error())
+                            .into_future();
+                    }
+                }
+                let claim_builder = ClaimsBuilder::default();
+                let claim = match claim_builder
+                    .with_user_id(&claims.user_id())
+                    .with_issuer(&rest_config.issuer())
+                    .with_duration(rest_config.access_token_duration())
+                    .build()
+                {
+                    Ok(claim) => claim,
+                    Err(err) => {
+                        error!("Failed to build claim {}", err);
+                        return HttpResponse::InternalServerError()
+                            .json(ErrorResponse::internal_error())
+                            .into_future();
+                    }
+                };
+
+                let token = match token_issuer.issue_token_with_claims(claim) {
+                    Ok(token) => token,
+                    Err(err) => {
+                        error!("Failed to issue token {}", err);
+                        return HttpResponse::InternalServerError()
+                            .json(ErrorResponse::internal_error())
+                            .into_future();
+                    }
+                };
+
+                HttpResponse::Ok()
+                    .json(json!({ "token": token }))
+                    .into_future()
+            }))
+        })
+}
+
+#[derive(Deserialize)]
+pub struct RefreshToken {
```

Comment:
This should probably get moved into the resources module, under `resources/token` to follow the pattern that's currently used